### ISSUE TITLE
Rouge-L score of pretraining with pt_16h is less than 40

### ISSUE THREAD

**Issue by andrewivan123** [PiotrNawrot/nanoT5#47]
I used the pt_16h.yaml config to do pretraining and then do fine-tuning afterwards using the ft.yaml.
Here is the pretraining evaluation result
`[2025-05-25 15:40:16,555][Main][INFO] - [eval] Step 53333 out of 53332 | Loss --> 1.873 | Accuracy --> 0.626 | Time --> 479.331 | `
Here is the finetuning evaluation result
`[2025-05-26 03:26:04,759][Main][INFO] - [eval] Step 18831 out of 18830 | Loss --> 1.333 | Accuracy --> 0.287 | Time --> 8.721 | 
[2025-05-26 03:34:32,292][absl][INFO] - Using default tokenizer.
[2025-05-26 03:34:35,971][Main][INFO] - [test] Step 18831 out of 18830 | Rougel --> 39.113 | Time --> 511.209 | `

As it can be seen, the Rouge-L score is less than 40.7 +- 2 . Is this normal? Are there any randomness involved inside the code?

default.yaml
`defaults:
  - _self_
  - task: pt
  - local_env: default

# Experiment args
mode: 'pt'
device: gpu
precision: 'bf16'
eval_only: false
predict_only: false
seed: 2137

model:
  klass: local_t5
  name: 'google/t5-v1_1-base'
  overwrite:
    dropout_rate: 0.0
  add_config:
    is_bf16: false
  checkpoint_path: ''
  random_init: true
  compile: false # Pytorch 2.0

data:
  input_length: 512
  mlm_probability: 0.15
  mean_noise_span_length: 3.0
  num_workers: 8

optim:
  name: adamwscale
  base_lr: 2e-2
  batch_size: 128
  total_steps: 65536
  epochs: -1 # If it's > 0 it overwrites total_steps
  warmup_steps: 10000
  lr_scheduler: cosine
  weight_decay: 0.0
  grad_clip: 1.0
  grad_acc: 1
  final_cosine: 1e-5

eval:
  every_steps: 100000 # Eval once in the end
  steps: 500

checkpoint:
  every_steps: 100000 # Save checkpoint once in the end

logging:
  neptune: false
  neptune_creds:
    project:
    api_token:
    tags: ''
  every_steps: 100
  grad_l2: true
  weights_l2: true

hydra:
  job:
    chdir: True
`

pt_16h.yaml
`# @package _global_

optim:
  warmup_steps: 6666
  batch_size: 256
  total_steps: 53332
  grad_acc: 4`
